nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/OperationChainContribution.java

```diff
@@ -207,16 +208,16 @@ public OperationChain toOperationChain(Bundle bundle) throws OperationException
                     case 'd':
                         if (T_DOCUMENT.equals(type)) {
                             if (param.value.startsWith(".")) {
-                                val = Scripting
-                                        .newExpression("Document" + ".resolvePathAsRef(\"" + param.value + "\")");
+                                val = Scripting.newExpression(
+                                        "Document" + ".resolvePathAsRef(\"" + param.value + "\")");
                             } else {
-                                val = StringToDocRef.createRef(param.value);
+                                val = param.value.startsWith("/") ? new PathRef(param.value) : new IdRef(param.value);
                             }
                         } else if (T_DOCUMENTS.equals(type)) {
                             String[] ar = StringUtils.split(param.value, ',', true);
                             DocumentRefListImpl result = new DocumentRefListImpl(ar.length);
                             for (String ref : ar) {
-                                result.add(StringToDocRef.createRef(ref));
+                                result.add(ref.startsWith("/") ? new PathRef(ref) : new IdRef(ref));
```

Comment:
Please move the old `StringToDocRef.createRef` to a `public static` method of `AbstractTypeAdapter` and use that (instead of duplicating code).